manim/renderer/shader.py

```diff
@@ -0,0 +1,272 @@
+import os
+import re
+from pathlib import Path
+
+import moderngl
+import numpy as np
+
+from .. import config
+from ..utils import opengl
+from ..utils.simple_functions import get_parameters
+
+SHADER_FOLDER = Path(__file__).parent / "shaders"
+shader_program_cache = {}
+file_path_to_code_map = {}
+
+
+def get_shader_code_from_file(file_path):
+    if file_path in file_path_to_code_map:
+        return file_path_to_code_map[file_path]
+    with open(file_path, "r") as f:
+        source = f.read()
+        include_lines = re.findall(r"^#include .*\.glsl$", source, flags=re.MULTILINE)
+        for line in include_lines:
+            include_path = line.split()[1]
```

Comment:
How about using regex groups rather than splitting the string.

```suggestion
        include_lines = re.finditer(r"^#include (?P<include_path>.*\.glsl)$", source, flags=re.MULTILINE)
        for line in include_lines:
            include_path = line.group("include_path")
```

Reply:
Done.